code/modules/medical/surgery_tools.dm

```diff
@@ -1148,8 +1150,18 @@ CONTAINS:
 		src.open()
 		src.visible_message("<span class='alert'><b>[user]</b> unzips themselves from [src]!</span>")
 
-	MouseDrop(mob/user as mob)
+	MouseDrop(atom/over_object)
+		if (!over_object) return
+		if(isturf(over_object))
+			..() //Lets it do the turf-to-turf slide
+			return
+		else if (istype(over_object, /obj/screen/hud))
+			MouseDrop(usr) //Try to fold & pick up the bag instead
+			return
+		else if (!ismob(over_object))
+			return
 		..()
+		var/mob/user = over_object
 		if (!(src.contents && src.contents.len) && (usr == user && !usr.restrained() && !usr.stat && in_range(src, usr) && !issilicon(usr)))
```

Comment:
```suggestion
		if (!length(src.contents) && usr.can_use_hands() && isalive(usr) && in_range(src, usr) && !issilicon(usr))
```
you didn't touch this line but holy shit this is gross. At least this is a little better, also removed some pointless checks